internal/pkg/service/common/etcdop/iterator/option.go

```diff
@@ -92,9 +96,40 @@ func WithFromSameRev(v bool) Option {
 	}
 }
 
-// WithEnd defines end of the iteration, all keys from the range [prefix/, prefix/end) will be loaded.
-func WithEnd(v string) Option {
+// WithStartOffset defines start of the iteration.
+// An empty string (default) means that the start is the first key in the prefix.
+// Iterated are all keys from the range (prefix/startOffset, prefix/endOffset).
+// Boundary values, start and end offsets, are excluded.
+func WithStartOffset(v string) Option {
 	return func(c *config) {
-		c.end = c.prefix + v
+		c.startOffset = v
 	}
 }
+
+// WithEndOffset defines end of the iteration.
+// An empty string (default) means that the end is the last key in the prefix.
+// Iterated are all keys from the range (prefix/startOffset, prefix/endOffset).
+// Boundary values, start and end offsets, are excluded.
+func WithEndOffset(v string) Option {
+	return func(c *config) {
+		c.endOffset = v
+	}
+}
+
+func (c config) start() string {
+	if c.startOffset != "" {
+		// Iterate from the startOffset, the startOffset is excluded.
+		return etcd.GetPrefixRangeEnd(c.prefix + c.startOffset)
+	}
+	// Iterate from the first key in the prefix.
+	return c.prefix
+}
+
+func (c config) end() string {
+	if c.endOffset != "" {
+		// Iterate to the endOffset, the endOffset is excluded.
+		return c.prefix + c.endOffset
+	}
+	// Iterate to the last key in the prefix.
+	return etcd.GetPrefixRangeEnd(c.prefix)
```

Comment:
Otherwise, iterate the whole prefix - calculate prefix end (it match old implementation, see next commits).